src/envoy/mixer/mixer_control.h

```diff
@@ -42,13 +66,19 @@ class HttpMixerControl final : public ThreadLocal::ThreadLocalObject {
 
   bool has_v2_config() const { return has_v2_config_; }
 
+  void StatsUpdateCallback();
+
  private:
   // Envoy cluster manager for making gRPC calls.
   Upstream::ClusterManager& cm_;
   // The mixer control
   std::unique_ptr<::istio::mixer_control::http::Controller> controller_;
   // has v2 config;
   bool has_v2_config_;
+
+  // These members are needed to update envoy stats periodically.
+  MixerStatsObject stats_;
+  std::unique_ptr<::istio::mixer_client::Timer> timer_;
```

Comment:
I think the timer logic can be moved into MixerStatsObj class.
It will take a function call in the contructor to call controller_->GetStatistitis()